packages/workflow/src/Interfaces.ts

```diff
@@ -2390,7 +2391,17 @@ export interface WorkflowTestData {
 }
 
 export type LogLevel = (typeof LOG_LEVELS)[number];
-export type Logger = Record<Exclude<LogLevel, 'silent'>, (message: string, meta?: object) => void>;
+export type LogMetadata = {
+	[key: string]: unknown;
+	scopes?: LogScope[];
+	file?: string;
+	function?: string;
+};
+export type Logger = Record<
+	Exclude<LogLevel, 'silent'>,
+	(message: string, metadata?: LogMetadata) => void
+>;
+export type LogLocationMetadata = Pick<LogMetadata, 'file' | 'function'>;
```

Comment:
1. Instead of picking from `LogMetadata` I wonder if it'd be more readable to define `LogLocationMetadata` as those two properties and intersect this type to create `LogMetadata`?
2. I'm not clear just from this what `Logger` is here as opposed to the `Logger` service. The proxy? Maybe we can docline this?

Reply:
I just copied this code over. I think we can refactor this bit in the next step when we get rid of more of the `LoggerProxy` code.